mantle/cmd/kola/testiso.go

```diff
@@ -402,113 +471,67 @@ func runTestIso(cmd *cobra.Command, args []string) error {
-		printResult(scenarioISOLiveLogin, duration, err)
-		if err != nil {
-			return err
+		if kola.HasString("mpath", tests) {
+			enableMultipath = true
 		}
-	}
-	if _, ok := targetScenarios[scenarioISOAsDisk]; ok {
-		if kola.CosaBuild.Meta.BuildArtifacts.LiveIso == nil {
-			return fmt.Errorf("build %s has no live ISO", kola.CosaBuild.Meta.Name)
+		if kola.HasString("secure", tests) {
+			enableUefiSecure = true
+		} else if kola.HasString("uefi", tests) {
+			 enableUefi = true
 		}
-		switch coreosarch.CurrentRpmArch() {
-		case "x86_64":
-			ranTest = true
-			duration, err := testAsDisk(ctx, filepath.Join(outputDir, scenarioISOAsDisk))
-			printResult(scenarioISOAsDisk, duration, err)
-			if err != nil {
-				return err
+
+		if kola.HasString("pxe", tests) {
+			checkIso("kernel")
+			if kola.HasString("offline", tests) {
+				ranTest = true
+				duration, err = testPXE(ctx, inst, filepath.Join(outputDir, pattern), true)
+			} else {
+				ranTest = true
+				duration, err = testPXE(ctx, inst, filepath.Join(outputDir, pattern), false)
+			}
+		} else {
+			checkIso("iso")
+			if kola.HasString("login", tests) {
+				ranTest = true
+				duration, err = testLiveLogin(ctx, filepath.Join(outputDir, pattern))
+			} else if strings.Contains(pattern, "disk") {
+				ranTest = true
+				duration, err = testAsDisk(ctx, filepath.Join(outputDir, pattern))
+			} else {
+				if kola.HasString("offline", tests) {
+					ranTest = true
+					duration, err = testLiveIso(ctx, inst, filepath.Join(outputDir, pattern), true, false)
+				} else {
+					ranTest = true
+					duration, err = testLiveIso(ctx, inst, filepath.Join(outputDir, pattern), false, false)
+				}
 			}
-		default:
-			// no hybrid partition table to boot from
-			fmt.Printf("%s unsupported on %s; skipping\n", scenarioISOAsDisk, coreosarch.CurrentRpmArch())
-		}
-	}
-	if _, ok := targetScenarios[scenarioMinISOInstall]; ok {
-		if kola.CosaBuild.Meta.BuildArtifacts.LiveIso == nil {
-			return fmt.Errorf("build %s has no live ISO", kola.CosaBuild.Meta.Name)
-		}
-		ranTest = true
-		instIso := baseInst // Pretend this is Rust and I wrote .copy()
-		duration, err := testLiveIso(ctx, instIso, filepath.Join(outputDir, scenarioMinISOInstall), false, true)
-		printResult(scenarioMinISOInstall, duration, err)
-		if err != nil {
-			return err
-		}
-	}
-	if _, ok := targetScenarios[scenarioMinISOInstallNm]; ok {
-		if kola.CosaBuild.Meta.BuildArtifacts.LiveIso == nil {
-			return fmt.Errorf("build %s has no live ISO", kola.CosaBuild.Meta.Name)
 		}
-		ranTest = true
-		instIso := baseInst // Pretend this is Rust and I wrote .copy()
-		addNmKeyfile = true
-		duration, err := testLiveIso(ctx, instIso, filepath.Join(outputDir, scenarioMinISOInstallNm), false, true)
-		printResult(scenarioMinISOInstallNm, duration, err)
+		printResult(pattern, duration, err)
 		if err != nil {
 			return err
 		}
-	}
 
+	}
 	if !ranTest {
 		panic("Nothing was tested!")
 	}
```

Comment:
looking at the new code I think `ranTest` can only be `false` if `checkIso()` returns an error and we error out.

Let's just drop `ranTest` completely and call `liveArtifactsExistInBuild()` up front and error out early if they don't.